File: Non Repeating Character[20-July-2023]/Solution.java
//  >> JAVA CODE <<

class Solution
{
    //Function to find the first non-repeating character in a string.
    static char nonrepeatingCharacter(String S)
    {
        int[] freq=new int[26];
        
        for(char ch:S.toCharArray())
        {
            freq[ch-'a']++;
        }
        
        for(char ch:S.toCharArray())
            if(freq[ch-'a']==1) return ch;
            
            return '$';
    }
}

File: Non Repeating Character[20-July-2023]/Solution.cpp
//  >> C++ CODE <<

class Solution
{
    public:
    //Function to find the first non-repeating character in a string.
    char nonrepeatingCharacter(string S)
    {
        vector<int> freq(26, 0);
    
        for (char ch : S)
        {
            freq[ch - 'a']++;
        }
    
        for (char ch : S)
        {
            if (freq[ch - 'a'] == 1)
                return ch;
		}
    
        return '$';
	}

};